File: dehaze-go/internal/model/vo/recommendation.go
package vo

// ColorDistribution 颜色分布
type ColorDistribution struct {
	Temperature float64 `json:"temperature"`
	Saturation  float64 `json:"saturation"`
}

// ImageFeatureAnalysisVO 图像特征分析结果
type ImageFeatureAnalysisVO struct {
	ImageMd5         string             `json:"imageMd5"`
	HazeLevel        string             `json:"hazeLevel"`
	HazeConfidence   float64            `json:"hazeConfidence"`
	SceneType        string             `json:"sceneType"`
	SceneConfidence  float64            `json:"sceneConfidence"`
	Lighting         string             `json:"lighting"`
	Complexity       float64            `json:"complexity"`
	ColorDistribution ColorDistribution `json:"colorDistribution"`
	Resolution       string             `json:"resolution"`
	NoiseLevel       string             `json:"noiseLevel"`
}

// RecommendedAlgorithmVO 推荐算法项
type RecommendedAlgorithmVO struct {
	RecommendationId  int64    `json:"recommendationId"`
	AlgorithmID       int64    `json:"algorithmId"`
	AlgorithmName     string   `json:"algorithmName"`
	MatchScore        int      `json:"matchScore"`
	Reason            string   `json:"reason"`
	Rating            *float64 `json:"rating"`
	EstimatedTime     *int     `json:"estimatedTime"`
	EffectDescription string   `json:"effectDescription"`
}

// RecommendationRuleVO 推荐规则
type RecommendationRuleVO struct {
	ID           int64    `json:"id"`
	RuleName     string   `json:"ruleName"`
	SceneType    string   `json:"sceneType"`
	AlgorithmIds []int64  `json:"algorithmIds"`
	Weight       int      `json:"weight"`
	Enabled      bool     `json:"enabled"`
}

// TrendItem 趋势数据项
type TrendItem struct {
	Date        string  `json:"date"`
	AdoptionRate float64 `json:"adoptionRate"`
}

// RecommendationReportVO 推荐效果报表
type RecommendationReportVO struct {
	TotalRecommendations int64       `json:"totalRecommendations"`
	AdoptionRate         float64     `json:"adoptionRate"`
	SatisfactionRate     float64     `json:"satisfactionRate"`
	CoverageRate         float64     `json:"coverageRate"`
	ColdStartSuccessRate float64     `json:"coldStartSuccessRate"`
	Trend                []TrendItem `json:"trend"`
}

File: dehaze-python/app/models/schema/recommendation.py
"""
推荐管理模块 Pydantic Schema
"""

from pydantic import BaseModel, Field


class ColorDistribution(BaseModel):
    temperature: float = Field(..., description="色温(K)")
    saturation: float = Field(..., description="饱和度(0-1)")


class ImageFeatureAnalysisVO(BaseModel):
    imageMd5: str | None = Field(default=None, description="图像MD5（用于后续查询推荐）")
    hazeLevel: str = Field(..., description="雾霾浓度(light/moderate/heavy)")
    hazeConfidence: float = Field(..., description="雾霾浓度置信度(0-1)")
    sceneType: str = Field(
        ..., description="场景类型(urban/landscape/building/night/backlight/indoor)"
    )
    sceneConfidence: float = Field(..., description="场景类型置信度(0-1)")
    lighting: str = Field(..., description="光照条件(bright/normal/dark/veryDark/backlight)")
    complexity: float = Field(..., description="图像复杂度(0-1)")
    colorDistribution: ColorDistribution = Field(..., description="颜色分布")
    resolution: str = Field(..., description="分辨率(sd/hd/uhd)")
    noiseLevel: str = Field(..., description="噪声水平(low/medium/high)")


class AnalyzeForm(BaseModel):
    imageId: int | None = Field(default=None, description="图片ID(imageId/imageUrl二选一)")
    imageUrl: str | None = Field(default=None, description="图片URL(imageId/imageUrl二选一)")


class RecommendedAlgorithmVO(BaseModel):
    recommendationId: int | None = Field(default=None, description="推荐记录ID（用于提交反馈）")
    algorithmId: int = Field(..., description="算法ID")
    algorithmName: str = Field(..., description="算法名称")
    matchScore: int = Field(..., description="匹配度(0-100)")
    reason: str = Field(..., description="推荐理由")
    rating: float | None = Field(default=None, description="算法评分(0-5)，待真实数据采集后填充")
    estimatedTime: int | None = Field(
        default=None, description="预估耗时(ms)，待真实数据采集后填充"
    )
    effectDescription: str = Field(..., description="预期效果描述")


class RecommendationFeedbackForm(BaseModel):
    recommendationId: int = Field(..., description="推荐记录ID")
    useful: bool = Field(..., description="是否有用")


class RecommendationRuleVO(BaseModel):
    id: int = Field(..., description="规则ID")
    ruleName: str = Field(..., description="规则名称")
    sceneType: str = Field(..., description="场景类型")
    algorithmIds: list[int] = Field(..., description="候选算法ID列表")
    weight: int = Field(..., description="规则权重(0-100)")
    enabled: bool = Field(..., description="是否启用")


class RecommendationRuleForm(BaseModel):
    id: int | None = Field(default=0, description="规则ID(0表示新增)")
    ruleName: str = Field(..., description="规则名称")
    sceneType: str = Field(..., description="场景类型")
    algorithmIds: list[int] = Field(..., description="候选算法ID列表")
    weight: int = Field(..., ge=0, le=100, description="规则权重(0-100)")
    enabled: bool = Field(default=True, description="是否启用")


class TrendItem(BaseModel):
    date: str = Field(..., description="日期")
    adoptionRate: float = Field(..., description="采纳率")


class RecommendationReportVO(BaseModel):
    totalRecommendations: int = Field(..., description="总推荐次数")
    adoptionRate: float = Field(..., description="采纳率")
    satisfactionRate: float = Field(..., description="满意度")
    coverageRate: float = Field(..., description="覆盖率")
    coldStartSuccessRate: float = Field(..., description="冷启动成功率")
    trend: list[TrendItem] = Field(..., description="趋势数据")


class IdVO(BaseModel):
    id: int = Field(..., description="ID")
